posthog/models/feature_flag/flag_matching.py

```diff
@@ -1023,7 +1024,7 @@ def get_all_feature_flags_with_details(
     try:
         # when we're writing a hash_key_override, we query the main database, not the replica
         # this is because we need to make sure the write is successful before we read it
-        using_database = "default" if writing_hash_key_override else DATABASE_FOR_FLAG_MATCHING
```

Comment:
yeah, so in this case `"default"` is the writer so we still were using the reader if we needed to write a hash key and we preferred the writer otherwise.